src/plugins/discover/public/application/main/data_fetching/fetch_all.ts

```diff
@@ -173,12 +185,17 @@ export function fetchAll(
       // but their errors will be shown in-place (e.g. of the chart).
       .catch(sendErrorTo(dataSubjects.documents$, dataSubjects.main$));
 
-    // Return a promise that will resolve once all the requests have finished or failed
+    // Return a promise that will resolve once all the requests have finished or failed, or no results are found
     return firstValueFrom(
-      combineLatest([
-        isComplete(dataSubjects.documents$).pipe(switchMap(async () => onFetchRecordsComplete?.())),
-        isComplete(dataSubjects.totalHits$),
-      ])
+      race(
+        combineLatest([
+          isComplete(dataSubjects.documents$).pipe(
+            switchMap(async () => onFetchRecordsComplete?.())
+          ),
+          isComplete(dataSubjects.totalHits$),
+        ]),
+        noResultsFound(dataSubjects.main$)
```

Comment:
This promise used to never resolve for searches that returned no results, and then the next search with results would resolve both of them. I don't think it had much of an impact since the `then` block is pretty simple, but better to handle it correctly regardless.